### PR TITLE
Mark SI pull requests job as failed.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
In some cases the launch would fail but we would not report it as such.

### CODE REVIEW

File: ci/si_pipeline.sh [hunk 1]
```diff
@@ -63,5 +63,5 @@ case $CLUSTER_LAUNCH_CODE in
       ;;
   2) exit-as-unstable "Cluster launch failed.";;
   3) exit-as-unstable "Cluster did not start in time.";;
-  *) echo "Unknown error in cluster launch"; exit "$CLUSTER_LAUNCH_CODE";;
+  *) exit-as-unstalbe "Unknown error in cluster launch: $CLUSTER_LAUNCH_CODE";;
```
Comment (alenkacz) [inline ci/si_pipeline.sh]: typo here :)

File: ci/si_pipeline.sh [hunk 2]
```diff
@@ -29,7 +29,7 @@ function create-junit-xml {
 }
 
 function exit-as-unstable {
-    echo "Cluster launch failed."
+    echo "$1"
     create-junit-xml "dcos-launch" "cluster.create" "$1"
     ./dcos-launch delete
     exit 0
```
Comment (alenkacz) [inline ci/si_pipeline.sh]: why do we exit with 0 here in case of error?

Comment (jeschkies) [reply ci/si_pipeline.sh]: Ah, because a launch failure should be an unstable build not a broken build.

Comment (alenkacz) [reply ci/si_pipeline.sh]: ok, makes sense! :) thanks

**Pull request merged**